lib/active_storage/attached/many.rb

```diff
@@ -7,14 +7,14 @@ class ActiveStorage::Attached::Many < ActiveStorage::Attached
   # You don't have to call this method to access the attachments' methods as
   # they are all available at the model level.
   def attachments
-    @attachments ||= ActiveStorage::Attachment.where(record_gid: record.to_gid.to_s, name: name)
+    @attachments ||= record.public_send("#{name}_attachments")
```

Comment:
Now the `@attachments` caching breaks `attachments.reload` for instance.

(Aside: I've been wanting a `association :some_poro` macro that could clear out the ivar on `reload` as well as maintain the reference on some_poro back to the model it's associated with)